sale_commission/wizard/wizard_settle.py

```diff
@@ -85,11 +85,15 @@ def action_settle(self):
                         settlement = settlement_obj.create(
                             {'agent': agent.id,
                              'date_from': sett_from,
-                             'date_to': sett_to})
+                             'date_to': sett_to,
+                             'company_id':
+                                agent_lines[pos].invoice_line.company_id.id})
```

Comment:
But the code for that is the same as the code for grouping by companies. If you don't want to do it, let me propose a PR to your branch, but it's a pity that we waste this occasion to make this real multi-company aware.

I know (and this is the advice I give), that companies that are parent of another (aka company groups) are only for viewing normally, but for example in a customer I have with a franchise model, this way of doing (using a company group) is the only way they have for invoicing for example franchises. For now, I'm not going to use commissions on them, but if so, we can let the work done.